doc/modules/cassandra/pages/new/configuration.adoc

```diff
@@ -0,0 +1,198 @@
+If the parameter is of type duration, data rate or data storage, its value should be accompanied by a unit when new name is used.
+
+- Please follow the new format `noun_verb` when adding new configuration parameters.
+
+- Please consider adding any new parameters with the lowest supported by Cassandra unit.
+
+- If for some reason you consider the smallest unit shouldn’t be the one that is supported as such in Cassandra, please,
+use the extended classes `SmallestDuration*`, `SmallestDataStorage*`.
+
+- New parameters should be added as non-negative numbers.
+
+- Any time you add @Replaces with a name change, we need to add an entry in this https://github.com/riptano/ccm/blob/808b6ca13526785b0fddfe1ead2383c060c4b8b6/ccmlib/common.py#L62[Python dictionary in CCM] to support the same backward compatibility as SnakeYAML.
+
+Please follow the instructions in requirements.txt in the DTest repo how to retag CCM after committing any changes.
+You might want to test also with tagging in your repo to ensure that there will be no surprise after retagging the official CCM.
+Please be sure to run a full CI after any changes as CCM affects a few of our testing suites.
+
+- Some configuration parameters are not announced in cassandra.yaml, but they are presented in the Config class for advanced users.
+Those also should be using the new framework and naming conventions.
+
+- As we have backward compatibility, we didn’t have to rework all python DTests to set config in the new format, and we exercise
+the backward compatibility while testing. Please consider adding any new tests using the new names and value format though.
+
+- In-JVM upgrade tests do not support per-version configuration at the moment, so we have to keep the old names and value format.
+Currently, if we try to use the new config for a newer version, that will be silently ignored and default config will be used.
+
+- SnakeYAML supports overloading of parameters. This means that if you add a configuration parameter more than once in your `cassandra.yaml` -
+the latest occasion will be the one to load in Config during Cassandra startup. In order to make upgrades as less disruptive as possible,
+we continue supporting that behavior also with adding old and new names of a parameter into `cassandra.yaml`.
+
+*Example*:
+
+If you add the following to `cassandra.yaml`:
+....
+hinted_handoff_enabled: true
+enabled_hinted_handolff: false
+....
+
+you will get loaded in `Config`:
+....
+hinted_handoff_enabled: false
+....
+
+https://issues.apache.org/jira/browse/CASSANDRA-17379[CASSANDRA-17379] was opened to improve the user experience and deprecate the overloading.
+The plan for improvement is:
+
+1. Fail startup if the duplicate parameters have different values.
+
+2. WARN on startup if duplicates exist but have the same value.
+
+3. Provide a system property to opt out of #1.
```

Comment:
I would probably just link to 17379, rather than fully summarize the implementation plan. One of these three points could change slightly...

Reply:
Good point, removed the plan from here